### DISCUSSION TITLE
How do I scrape springboot actuator health?

### DISCUSSION THREAD

**Discussion by govardha** [signalfx/splunk-otel-collector-chart#667] [Q&A] [answered]
Hello,

The micro services I deploy in my k8s cluster have the spring boot actuators that expose prom metrics + health(live Ness/readiness etc).  

Scraping Prometheus metric seems straight forward enough(I haven’t tried it yet, but I suspect I will get it done), but is there a way to probe health.  The results comes back as a json payload {status: “UP”, key=value}.  

Or am I going about this the wrong way and I should instrument the micro service to directly inform the otel agent about its status?

Thank you!
G

**Comment by atoulme** [comment, marked as answer]
You would do this in the collector by setting up the prometheus receiver. See https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver for reference.

**Reply by govardha** [reply]
I got this working and it works like beautifully.  There is a helm option in the `agent` section where you enable prometheus.scraping(it is set to false by default).  This introduces a `receiver_creator` which looks for annotations that indicate whether prometheus scraping is available and the port & end point to scrape.

I introduced these annotations and then tweaked the pipeline to drop certain metrics which were not useful and the `otelcol` starts scraping this perfectly and ships it over to the Splunk IDX.
